hack/build_multiarch_clis.sh

```diff
@@ -63,4 +93,50 @@ for GOARCH in amd64 arm64; do
   done
 done
 
+echo -e "\033[0;37m>>>> building index.json\033[0m"
+# BASE_URL should be set in environment
+SOURCE_DIR="bin/clis/gh-archives"
+OUTPUT_JSON="index.json"
+
+cd $SOURCE_DIR
+
+echo '{"links": [' > $OUTPUT_JSON
+find . -type f | while read -r file; do
+  # Extract architecture and OS from path
+  arch=$(echo "$file" | awk -F'/' '{print $2}')
+  os=$(echo "$file" | awk -F'/' '{print $3}')
+  filename=$(echo "$file" | awk -F'/' '{print $NF}')
+
+  # Skip an index.json file
+  if [[ "$filename" == "$OUTPUT_JSON" ]]; then
+      continue
+  fi
+
+  # Skip a SHA256 file
+  if [[ "$filename" == *.sha256 ]]; then
+      continue
+  fi
+
+  # Corresponding SHA256 file
+  sha256_file="${file}.sha256"
+  if [[ -f "$sha256_file" ]]; then
+      sha256_file="${filename}.sha256"
+  else
+      sha256_file=""
+  fi
+
+  # Construct JSON entry
+  echo "  {" >> $OUTPUT_JSON
+  echo "    \"os\": \"$os\"," >> $OUTPUT_JSON
+  echo "    \"arch\": \"$arch\"," >> $OUTPUT_JSON
+  echo "    \"url\": \"$BASE_URL/$filename\"," >> $OUTPUT_JSON
+  echo "    \"sha256\": \"$BASE_URL/$sha256_file\"" >> $OUTPUT_JSON
```

Comment:
this feels slightly brittle, could we consider jq -n to generate the json?

Reply:
replaced